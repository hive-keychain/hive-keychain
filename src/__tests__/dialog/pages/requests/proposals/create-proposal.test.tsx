import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import CreateProposal from 'src/dialog/pages/requests/proposals/create-proposal';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestProposals from 'src/__tests__/utils-for-testing/data/props/dialog/request-proposals';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
describe('create-proposal tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestProposals.create;
  methods.config();
  it('Must show create proposal dialog', async () => {
    const { asFragment } = render(<CreateProposal {...props} />);
    expect(asFragment()).toMatchSnapshot('Request CreateProposal with loading');
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<CreateProposal {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    render(<CreateProposal {...{ ...props, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    render(<CreateProposal {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
