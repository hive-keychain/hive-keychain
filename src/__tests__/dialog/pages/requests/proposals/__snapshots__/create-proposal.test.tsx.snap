// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-proposal tests:
 Must show create proposal dialog: Request CreateProposal with loading 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Create Proposal
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Receiver
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain
      </div>
      <h3>
        Title
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        Create a keychain coin
      </div>
      <h3>
        Permlink
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        http://hive-keychain.com/coin
      </div>
      <h3>
        Period
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        21/12/2022 - 21/12/2024
      </div>
      <h3>
        Daily Pay
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        300 HBD
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
